First Draft

dhorki opened the pull request:
Hi @Apostat23

Mind I if ask you to take a look to the project?
Please review the project as a whole:

Delivered:
- Runnable application
- Database model definitions
- Seeds to populate DB with custom pokemons
- Way to get the data from Api and push it to Database
- Dashboard to see all pokemons
- Form to create/edit pokemons
- Database and model validations
- Authentication

There are some missing features like finished tests with RSpec

There should be massive opportunities to improve it, all kind of comments are welcomed

Thanks in advance

* as a side note, I was working directly with the main branch of the fork, just then I realized I should have used another branch for the development, once this pull request is completed I would continue to use this pokeapi_app branch and send the next pull requests pointed to the main branch

dhorki committed:
Added Pokemon App draft

dhorki committed:
Updated Index to mention the view corresponds to a catalog
Added link to admin menu if logged

dhorki committed:
Removed unnecessary files

dhorki committed:
Update seed

dhorki committed:
Added password confirmation to user
Corrected views

dhorki committed:
Some rubocop fixes
Navigation params

dhorki committed:
-Added validation to the pokeapi loader to avoid re-submitting pokemons
-Corrected Item empty spaces in code

dhorki committed:
Updated migrations field definition

dhorki committed:
Made loader available to logged users only
updated schema

dhorki committed:
Updated pages title/header

Apostat23 commented on pokemon_app/app/controllers/abilities_controller.rb:
```diff
@@ -0,0 +1,68 @@
+      page: params[:page]
+    }
+  end
+end
```
An extra empty line at the end of each file is taken as a good practice.

Apostat23 commented on pokemon_app/app/controllers/abilities_pokemons_controller.rb:
```diff
@@ -0,0 +1,77 @@
+      ability_page: @ability_page
+    }
+  end
+end
```
I think your IDE erases that extra empty lines if exists...

Apostat23 commented on pokemon_app/app/controllers/access_controller.rb:
```diff
@@ -0,0 +1,41 @@
+      session[:user_id] = authorized_user.id
+      session[:username] = authorized_user.username
+      flash[:notice] = 'You are now logged in.'
+      redirect_to(access_menu_path)
```
TRy to avoid if possible the use of `()`. I would set this as `redirect_to access_menu_path`

Apostat23 commented on pokemon_app/app/controllers/access_controller.rb:
```diff
@@ -0,0 +1,41 @@
+    flash[:notice] = 'Logged out'
+    redirect_to(access_login_path)
+  end
+end
```
Nice going :).... o good option too may be the Devise gem. Your way is perfect to understand the how's, I still would use the gem.

Apostat23 commented on pokemon_app/app/controllers/moves_pokemons_controller.rb:
```diff
@@ -0,0 +1,77 @@
+  layout 'admin'
+
+  before_action :move_page
+  before_action :navigation_params
```
I am not sure, try to set both methods in just one line, do not remember if it is possible or not, but it may worth to try.

dhorki committed:
application.css - removed extension to main.css file require
json response for pokemon index and show
refined validation for height and weight fields in the Pokemon model
changed to number_field the numeric attributes in the forms

dhorki committed:
Added heroku buildpath files

dhorki committed:
Copied project.json to root

dhorki committed:
Added x86_64-linux platform for heroku push

dhorki committed:
Merge branch 'heroku_version' of https://github.com/dhorki/ruby-on-rails-learning-path into heroku_version

dhorki committed:
Deleted past attempt files for heroku

dhorki committed:
corrected linux platform addition

dhorki committed:
Added favicon
Added url validation to sprites urls

dhorki committed:
Merge branch 'heroku_version'
